client/src/components/item/AddOptionDropdown.tsx

```diff
@@ -1,65 +1,78 @@
-import React, {Component} from 'react';
-import {Dropdown} from "semantic-ui-react";
+import React, {Component} from "react";
+import {Dropdown, DropdownProps} from "semantic-ui-react";
 
 interface AddOptionDropdownProps {
-    name: string,
-    required: boolean,
-    placeholder: string,
-    options: Option[]
+    name: string;
+    required: boolean;
+    placeholder: string;
+    options: Option[];
+    loading?: boolean;
+    disabled?: boolean;
+    error?: boolean;
+    value?: string;
+    onChange: (event: React.SyntheticEvent<HTMLElement>, data: DropdownProps, inputName: string) => void;
 }
 
 export interface Option {
-    text: string,
-    value: string
+    text: string;
+    value: string;
 }
 
 interface AddOptionDropdownState {
-    options: Option[],
-    currentValue: string
+    options: Option[];
+    currentValue?: any;
 }
 
 class AddOptionDropdown extends Component<AddOptionDropdownProps, AddOptionDropdownState> {
     constructor(props: AddOptionDropdownProps) {
         super(props);
-        this.handleAddition = this.handleAddition.bind(this);
-        this.handleChange = this.handleChange.bind(this);
+        console.log("props are:", props.options);
         this.state = {
             options: this.props.options,
-            currentValue: ""
+            currentValue: this.props.value || ""
         };
     }
 
-    handleAddition = (e: any, {value}: any) => {
+    public handleAddition = (e: any, {value}: any) => {
         this.setState({
             options: [{text: value, value}, ...this.state.options],
-        })
-    };
+        });
+    }
 
-    handleChange = (e: any, {value}: any) => this.setState({currentValue: value});
+    public handleChange = (event: React.SyntheticEvent<HTMLElement>, data: DropdownProps) => {
+        console.log(event, data.value);
+        this.setState({
+            currentValue: data.value
+        });
+        this.props.onChange(event, data, this.props.name);
+    }
     // onAddItem(e: React.KeyboardEvent<HTMLElement>, dropdown: DropdownProps): void {
     //     const value: string = dropdown.
     //     this.setState({
     //         options: [{ text: value, value: value }, ...this.state.options],
     //     });
     // }
 
-    render() {
+    public render() {
         return (
             <Dropdown width={4}
                       type="text"
                       name={this.props.name}
                       allowAdditions
                       onAddItem={this.handleAddition}
+                      error={this.props.error || false}
```

Comment:
|| false? Is this some js thing?

Reply:
Means if the first operand is undefined it'll use the second value, tbh not sure if it's 100% necessary but it's not gonna break anything ™

Reply:
Turns out it's unnecessary, so I'll remove it